### PR TITLE
Update Ruby version from 3.0.5 to 3.1.6

### PR DESCRIPTION
**Pull request by jayjay-w**
## Description

Upgrade Pender's ruby version from 3.0.5 to 3.1.6. 

This is the first step in a multi-step process which will be done in steps: 3.0.5 -> 3.16 -> 3.2.5 -> 3.3.4

3.1.6 will be supported until 2025-03-31 (expected), so this will ensure that we are using
a ruby version that is maintained and up to date.

References: CV2-4957

## How has this been tested?

Ensured that all tests are passing

## Things to pay attention to during code review

All functionality should be working as expected.

## Checklist

- [x] I have performed a self-review of my own code
- [ ] I have added unit and feature tests, if the PR implements a new feature or otherwise would benefit from additional testing
- [ ] I have added regression tests, if the PR fixes a bug
- [ ] I have added logging, exception reporting, and custom tracing with any additional information required for debugging
- [x] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [ ] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [ ] My changes generate no new warnings
- [ ] If I added a third party module, I included a rationale for doing so and followed our current [guidelines](https://meedan.atlassian.net/wiki/spaces/ENG/overview#Choose-the-%E2%80%9Cright%E2%80%9D-3rd-party-module)

### CODE REVIEW

File: production/Dockerfile [hunk 1]
```diff
@@ -22,7 +22,7 @@ ENV DEPLOYUSER=pender \
 COPY production/bin /opt/bin/
 RUN chmod 755 /opt/bin/*.sh
 
-RUN apt-get update && apt-get install -y curl build-essential git libpq-dev graphicsmagick inotify-tools python --no-install-recommends
+RUN apt-get update && apt-get install -y curl build-essential git libpq-dev graphicsmagick inotify-tools --no-install-recommends
```
[the remark this answers is not in the document]
Comment (caiosba) [reply production/Dockerfile:25]: Yes, I think @vasconsaurus is doing the same in her branch (Manu, please confirm).

File: test/integration/parsers/facebook_item_test.rb [hunk 2]
```diff
@@ -1,28 +1,28 @@
...
-    assert !data['description'].blank?
-    assert !data['text'].blank?
-    assert !data['picture'].blank?
-    assert !data['published_at'].blank?
-    # data['html'] started to be returned as an empty string for this test
-    # which is extra weird since we get it even when the page does not exist
-    # will come back to this
-    # assert !data['html'].blank?
-  end
+  # test "should get facebook post with valid data from crowdtangle" do
```
[the remark this answers is not in the document]
Comment (caiosba) [reply test/integration/parsers/facebook_item_test.rb:4]: So I assume you're tracking that in CV2-4384 right @jayjay-w ?

**Review by caiosba — APPROVED** [review]

**Comment by caiosba** [comment]
@jayjay-w please make sure integration tests in Check Web pass against this branch before merging.

File: production/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (vasconsaurus) [reply production/Dockerfile:25]: Yes, from what I looked into it seems to indeed only have been used by youtube-dl, so I'm are also removing it.

**Review by vasconsaurus — APPROVED** [review]

**Pull request merged**